apps/mobile/src/components/splash-screen-guard/splash-screen-guard.tsx

```diff
@@ -1,23 +1,144 @@
-import { useState } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useColorScheme } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
 
+import { useAppState } from '@/hooks/app-state';
+import { useSettings } from '@/store/settings/settings';
 import { HasChildren } from '@/utils/types';
+import { t } from '@lingui/macro';
+import { useTheme } from '@shopify/restyle';
+import * as LocalAuthentication from 'expo-local-authentication';
+import * as SplashScreen from 'expo-splash-screen';
+import * as SystemUI from 'expo-system-ui';
+import LottieView from 'lottie-react-native';
 
-import { Box } from '@leather.io/ui/native';
+import { colorThemes } from '@leather.io/tokens';
+import { Box, Button, Theme } from '@leather.io/ui/native';
 
+import { LeatherLockedSplash } from '../animations/leather-locked-splash';
 import { LeatherSplash } from '../animations/leather-splash';
 
-const DEFAULT_ANIMATION_FINISHED = __DEV__;
+const DEFAULT_ANIMATION_FINISHED = false; // __DEV__;
+type AuthState = 'started' | 'failed' | 'passedOnFirst' | 'passedAfterwards';
 
 export function SplashScreenGuard({ children }: HasChildren) {
   const [animationFinished, setAnimationFinished] = useState(DEFAULT_ANIMATION_FINISHED);
+  const splashRef = useRef<LottieView>(null);
+  const lockedSplashRef = useRef<LottieView>(null);
+  const { securityLevelPreference, whenTheme } = useSettings();
+  const [authState, setAuthState] = useState<AuthState>('started');
+  const insets = useSafeAreaInsets();
+  const theme = useTheme<Theme>();
+  const colorScheme = useColorScheme();
 
-  if (!animationFinished) {
-    return (
-      <Box backgroundColor="ink.component-background-default" flex={1}>
-        <LeatherSplash onAnimationEnd={() => setAnimationFinished(true)} />
-      </Box>
+  useEffect(() => {
+    // just in case splash screen flashes, we want to at least have the correct colors set
+    void SystemUI.setBackgroundColorAsync(
+      colorScheme === 'dark'
+        ? theme.colors['ink.text-non-interactive']
+        : theme.colors['ink.text-primary']
```

Comment:
There's a `whenTheme` helper to avoid the conditional
```suggestion
      whenTheme(colorScheme)(
        dark: theme.colors['ink.text-non-interactive'],
        light: theme.colors['ink.text-primary']
      )
```